lib/web/machineid.go

```diff
@@ -262,6 +263,45 @@ type updateBotRequest struct {
 	Roles []string `json:"roles"`
 }
 
+// getBotInstance retrieves a bot instance by id
+func (h *Handler) getBotInstance(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (any, error) {
+	botName := p.ByName("name")
+	instanceId := p.ByName("id")
+	if botName == "" {
+		return nil, trace.BadParameter("empty bot name")
+	}
+	if instanceId == "" {
+		return nil, trace.BadParameter("empty id")
+	}
+
+	clt, err := sctx.GetUserClient(r.Context(), site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	instance, err := clt.BotInstanceServiceClient().GetBotInstance(r.Context(), &machineidv1.GetBotInstanceRequest{
+		InstanceId: instanceId,
+		BotName:    botName,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err, "error querying bot instance")
+	}
+
+	yaml, err := yaml.Marshal(instance)
```

Comment:
I think this yaml.Marshal may under the hood be calling the "wrong" JSON marshalling function - which is why we see a weird object for the timestamps rather than a human readable timestamp.

My gut feeling is that if you try wrapping instance in `types.ProtoResource153ToLegacy(instance)` before passing it to yaml.Marshal, this will probably select the "correct" JSON marshaller instead.